pkg/buildpack/builder.go

```diff
@@ -330,13 +330,22 @@ func (b *PackageBuilder) validate() error {
 
 func (b *PackageBuilder) resolvedStacks() []dist.Stack {
 	stacks := b.buildpack.Descriptor().Stacks()
+	if len(stacks) == 0 && len(b.buildpack.Descriptor().Order()) == 0 {
+		// For non-meta-buildpacks using targets, not stacks: assume any stack
+		stacks = append(stacks, dist.Stack{ID: "*"})
```

Comment:
Should we need to add `b.buildpack.Descriptor().API().AtLeast("0.10")` to the condition?

Reply:
My opinion: no. I'm happy to do so if y'all want; just let me know. My reasoning: old versions of the Buildpack API spec do not specify that `[[stacks]]` is required or mandatory. It seems like `[[stacks]]` has been mandatory in `pack` because of this logic, and perhaps that was unintentional.

Reply:
> Should we need to add b.buildpack.Descriptor().API().AtLeast("0.10") to the condition?

I would be okay leaving this out. Older buildpacks should be able to work on newer platforms (where stacks are not required) without having to make changes.